tickit/adapters/interpreters/wrappers/splitting_interpreter.py

```diff
@@ -0,0 +1,110 @@
+
+from tickit.adapters.interpreters.utils import wrap_as_async_iterable
+from tickit.core.adapter import Adapter, Interpreter
+
+
+class SplittingInterpreter(Interpreter[AnyStr]):
+    """An wrapper for an interpreter that splits a single message into multiple.
+
+    An interpreter wrapper class that takes a message, splits it according to a given
+    delimiter, and passes on the resulting sub-messages individually on to the
+    wrapped interpreter. The individual responses to the sub-messages are combined into
+    a single response.
+    """
+
+    @overload
+    def __init__(self, interpreter: Interpreter[AnyStr]) -> None:  # noqa: D107
+        pass
+
+    @overload
+    def __init__(
+        self, interpreter: Interpreter[AnyStr], delimiter: AnyStr
+    ) -> None:  # noqa: D107
+        pass
+
+    def __init__(self, interpreter: Interpreter[AnyStr], delimiter=b" ") -> None:
+        """A decorator for an interpreter that splits a message into multiple sub-messages.
+
+        Args:
+            interpreter (Interpreter): The interpreter messages are passed on to.
+            delimiter (AnyStr): The delimiter by which the message is split up.
+
+        """
+        super().__init__()
+        self.interpreter: Interpreter[AnyStr] = interpreter
+        self.delimiter: AnyStr = delimiter
+
+    async def _handle_individual_messages(
+        self, adapter: Adapter, individual_messages: List[AnyStr]
+    ) -> List[Tuple[AsyncIterable[AnyStr], bool]]:
+        results = [
+            await self.interpreter.handle(adapter, message)
+            for message in individual_messages
+        ]
+        return results
+
+    @staticmethod
+    async def _get_response_and_interrupt_from_individual_results(
+        results: List[Tuple[AsyncIterable[AnyStr], bool]]
+    ) -> Tuple[AsyncIterable[AnyStr], bool]:
+        """Combines results from handling multiple messages.
+
+        Takes the responses from when the wrapped interpreter handles multiple messages
+        and returns an appropriate composite repsonse and interrrupt. The composite
+        response is the concatentation of each of the individual responses, the
+        composite interrupt is a logical inclusive 'or' of all of the individual
+        responses.
+
+        Args:
+            results (List[Tuple[AsyncIterable[AnyStr], bool]]): a list of returned
+                values from the wrapped class' handle() method.
+
+        Returns:
+            Tuple[AsyncIterable[AnyStr], bool]:
+                A tuple of the asynchronous iterable of reply messages and a flag
+                indicating whether an interrupt should be raised by the adapter.
+        """
+        individual_responses, individual_interrupts = zip(*results)
+
+        response_list = [
+            response
+            for response_gen in individual_responses
+            async for response in response_gen
+        ]
+        response = functools.reduce(lambda a, b: a + b, response_list)
```

Comment:
We did consider this. Whilst it's conceivable that a response where the individual messages are separated in some specific format could be required, we can't really think of one; so perhaps just add this feature if/when it's needed? We also would have to ensure that both delimiters are passed in as the same type - how much complication could this add?

Reply:
I don't think it would add any complication if the line were literally

```python
response = "".join(response_list)
```

which, I agree with @garryod, is more readable. You could also have 

```python
response = sum(response_list)
```

I generally use `functools.reduce` when there aren't alternatives like this.

Reply:
I think `response = "".join(response_list)` may introduce typing issues since responses are of type `AnyStr` and we would need `b"".join(...)` sometimes?

Also I don't think `sum` works with `str` or `bytes`?

Reply:
`sum` is literally `functools.reduce(lambda a, b: a + b, collection)`, but I think that may be less readable in this case.
I think you're right about `join` too, so maybe leave as is

Reply:
Afaik `sum` cannot be used for string types, even with `start="some_str"`. However, the fact that `reduce(lambda a, b: a + b, response_list)` is allowed by mypy is actually erroneous for `response_list: list[Union[str, bytes]]` as there is no implementation of `__add__(self, other)` for `self: str` & `other: bytes` or vice-versa, as such, we would need to type the input as `Union[list[str], list[bytes]]` or use a `TypeVar` to achieve the same (preferable), at which point ensuring both deliminators and the messages are the same type is fairly trivial.

Reply:
@callumforrester 

```Python console
>>> sum(["a", "b", "c"])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for +: 'int' and 'str'
>>> sum(["a", "b", "c"], start="")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: sum() can't sum strings [use ''.join(seq) instead]
```

Reply:
My bad...